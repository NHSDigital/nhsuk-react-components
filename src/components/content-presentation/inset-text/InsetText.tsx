import React, { HTMLProps } from 'react';
import classNames from 'classnames';

interface InsetTextProps extends HTMLProps<HTMLDivElement> {
  visuallyHiddenText?: string | false;
}

const InsetText: React.FC<InsetTextProps> = ({
  className,
  children,
  visuallyHiddenText,
  ...rest
}) => (
  <div className={classNames('nhsuk-inset-text', className)} {...rest}>
    {visuallyHiddenText ? (
      <span className="nhsuk-u-visually-hidden">{visuallyHiddenText}</span>
    ) : null}
    {children}
  </div>
);

InsetText.defaultProps = {
  visuallyHiddenText: 'Information: ',
};

export default InsetText;
