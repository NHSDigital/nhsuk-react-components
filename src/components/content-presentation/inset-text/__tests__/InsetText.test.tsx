import React from 'react';
import { shallow } from 'enzyme';
import InsetText from '..';

describe('InsetText', () => {
  it('matches snapshot', () => {
    const component = shallow(<InsetText />);
    expect(component).toMatchSnapshot('InsetText');
    component.unmount();
  });

  it('has default visually hidden text', () => {
    const component = shallow(<InsetText />);
    expect(component.find('.nhsuk-u-visually-hidden').text()).toBe('Information: ');
    component.unmount();
  });

  it('has disabled visually hidden text', () => {
    const component = shallow(<InsetText visuallyHiddenText={false} />);
    expect(component.find('.nhsuk-u-visually-hidden').exists()).toBeFalsy();
    component.unmount();
  });

  it('has custom visually hidden text', () => {
    const component = shallow(<InsetText visuallyHiddenText="Custom" />);
    expect(component.find('.nhsuk-u-visually-hidden').text()).toBe('Custom');
    component.unmount();
  });

  it('renders children', () => {
    const component = shallow(<InsetText>Child</InsetText>);
    expect(component.text()).toBe('Information: Child');
    component.unmount();
  });
});
