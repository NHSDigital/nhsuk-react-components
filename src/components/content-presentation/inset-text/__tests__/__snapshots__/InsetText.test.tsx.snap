// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsetText matches snapshot: InsetText 1`] = `
<div
  className="nhsuk-inset-text"
>
  <span
    className="nhsuk-u-visually-hidden"
  >
    Information: 
  </span>
</div>
`;
